### DISCUSSION TITLE
Skip single endpoint for http

### DISCUSSION THREAD

**Discussion by indapublic** [nginx-proxy/nginx-proxy#1894] [Q&A] [answered]
Hi all! First, thanks for your great product, using it for years.
I have a case when I need to have 1 single http endpoint in my https website behind nginx-proxy (it's requirement and can not be undone unfortunately). It can be somehow implemented out of box with `nginx-proxy`? Thanks in advance

**Comment by buchdag** [comment, marked as answer]
Do you mean redirecting HTTP to HTTPS for all the requests but still allow HTTP for one specific location ?

**Reply by indapublic** [reply]
Correct

**Reply by buchdag** [reply]
That won't be possible out of the box, but you can always customise the existing `nginx.tmpl`

The block you'll have to modify is this one https://github.com/nginx-proxy/nginx-proxy/blob/fb8ddfd08c0f6cb53e583ea22ff8be06f69c50dc/nginx.tmpl#L313-L343

For instance something like this

```diff
{{ if eq $https_method "redirect" }}
server {
	server_name {{ $host }};
	{{ if $server_tokens }}
	server_tokens {{ $server_tokens }};
	{{ end }}
	listen {{ $external_http_port }} {{ $default_server }};
	{{ if $enable_ipv6 }}
	listen [::]:{{ $external_http_port }} {{ $default_server }};
	{{ end }}
	{{ $access_log }}

	# Do not HTTPS redirect Let'sEncrypt ACME challenge
	location ^~ /.well-known/acme-challenge/ {
		auth_basic off;
		auth_request off;
		allow all;
		root /usr/share/nginx/html;
		try_files $uri =404;
		break;
	}
	
+	{{ if eq $host "www.example.com" }}
+      location /http-endpoint {
+		proxy_pass http://$upstream_name/;
+	}
+	{{ end }}
+
	location / {
		{{ if eq $external_https_port "443" }}
		return 301 https://$host$request_uri;
		{{ else }}
		return 301 https://$host:{{ $external_https_port }}$request_uri;
		{{ end }}
	}
}
{{ end }}
```

Should add a non redirected `/http-endpoint` location only when the `VIRTUAL_HOST` is `www.example.com`. I'm not 100% sure if the `proxy_pass` should be `http://$upstream_name/` or `http://$upstream_name$request_uri` but you get the idea.

**Reply by indapublic** [reply]
Awesome, I will try that and inform. Thank you!

**Reply by indapublic** [reply]
This section appears in `http` section after building, `https` section still contains only one location.

**Reply by buchdag** [reply]
If you want say incoming requests for `https://www.example.com/endpoint` to be proxied to `http://yourcontainer/endpoint`, there is nothing to add, that's the default behaviour. The only thing that need to be customised in this case is the bypass of the `HTTP` to `HTTPS` redirection for this specific location (`/endpoint`), as shown above.

If you need to customise the `server` block that correspond to redirected `HTTPS`, it's that one : https://github.com/nginx-proxy/nginx-proxy/blob/fb8ddfd08c0f6cb53e583ea22ff8be06f69c50dc/nginx.tmpl#L345-L410

**Reply by indapublic** [reply]
Ah, sorry, seems it was HSTS. Works. Thank you

**Reply by indapublic** [reply]
Does using the full nginx template mean that in theory something might go wrong when the library will be updated?

**Reply by buchdag** [reply]
Things shouldn't go unexpectedly wrong but using a custom `nginx.tmpl` file means you won't get upstream fixes and new features when updating the image, because almost all of them come from the template. You'll have to grab the up to date [upstream file](https://raw.githubusercontent.com/nginx-proxy/nginx-proxy/main/nginx.tmpl) from time to time, add your customisation again, then use it inside the nginx-proxy container.

I would advise you to version your custom `nginx.tmpl` files on your host with the corresponding release tag or commit hash.
